api/lib/sendGrid/sendGrid.js

```diff
@@ -0,0 +1,45 @@
+const { key } = require( "../keys/emailKey.json" );
+const logger = require( "@starryinternet/jobi" );
+const mongoose    = require( "mongoose" );
+const User     = require( "../models/user" );
+const sgMail = require( "@sendgrid/mail" );
+const JWTUtils  = require( "../utils/jwt" );
+const templates = require( "./emailTemplates/templates" );
+
```

Comment:
We want to use a class in `lib/classes/send-grid` that has the `sendWelcomeEmail` function in it (as well as some others). Creating a module like this and exporting from it is pretty close to creating a class, the only difference is that a class will allow us to have private properties and to construct the `SendGrid` instance with the key when the class is created (at app start) then delete the key.

For example in `lib/classes/sendgrid`:
```js
class SendGrid {
  #apiKey // this is a private property that will not be able to be accessed outside of this class definition

  constructor( opts ) {
    this.#apiKey = opts.key;

    this.defaultEmail = opts.defaultEmail;

    console.log( this.#apiKey );

    sgMail.setAPIKey( this.#apiKey );
    this.sendGrid = sgMail;
  }

  sendSomeTypeofMessage( recipients, subject, message ) {
    this.sendGrid.send({
      to: recipients,
      from: this.defaultEmail,
      subject,
      message
    });
  }
}
```

Then in `lib/sendGrid` we instantiate an instance with the key and delete the key:
note: class definitions are all caps wile instances are are not capitalized eg: `SendGrid` vs `sendGrid`.

```

const key =  getKey(); // could prompt for pass phrase or use another way to authenticate before decrypting key

module.exports = new SendGrid({ 
  key,
  defaultEmail: 'meetingminder@meetingminder.io` 
});

```
This will allow us to keep the key safe and put it up on gh encrypted.